iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -544,14 +549,20 @@ private void commitWrite(Table table, String branchName, Long snapshotId, long s
    * @param rewritePolicy The rewrite policy to use for the insert overwrite commit
    */
   private void commitOverwrite(Table table, String branchName, long startTime, FilesForCommit results,
-      RewritePolicy rewritePolicy) {
+      RewritePolicy rewritePolicy, String partitionSpec) {
     Preconditions.checkArgument(results.deleteFiles().isEmpty(), "Can not handle deletes with overwrite");
     if (!results.dataFiles().isEmpty()) {
       Transaction transaction = table.newTransaction();
       if (rewritePolicy == RewritePolicy.ALL_PARTITIONS) {
         DeleteFiles delete = transaction.newDelete();
         delete.deleteFromRowFilter(Expressions.alwaysTrue());
         delete.commit();
+      } else if (rewritePolicy == RewritePolicy.SINGLE_PARTITION) {
```

Comment:
should it be `rewritePolicy != RewritePolicy.ALL_PARTITIONS`?

Reply:
If it were `rewritePolicy != RewritePolicy.ALL_PARTITIONS`, it would also be true for the case `rewritePolicy != RewritePolicy.DEFAULT`, but the code in the else block should only be executed for compacting a specific partition, not in general case.

Reply:
can't we use the DEFAULT? rewrite policy enum is either all or not all (custom set)

Reply:
done